purchases/src/main/kotlin/com/revenuecat/purchases/caching/DeviceCache.kt

```diff
@@ -10,36 +10,42 @@ import com.revenuecat.purchases.Offerings
 import com.revenuecat.purchases.PurchaseWrapper
 import com.revenuecat.purchases.PurchaserInfo
 import com.revenuecat.purchases.Purchases
+import com.revenuecat.purchases.attributes.SubscriberAttribute
 import com.revenuecat.purchases.buildPurchaserInfo
 import com.revenuecat.purchases.debugLog
 import com.revenuecat.purchases.sha1
-
 import org.json.JSONException
 import org.json.JSONObject
 import java.util.Date
 
 private const val CACHE_REFRESH_PERIOD = 60000 * 5
-private const val PREFERENCES_PREFIX = "com.revenuecat.purchases."
+private const val SHARED_PREFERENCES_PREFIX = "com.revenuecat.purchases."
 
 internal class DeviceCache(
     private val preferences: SharedPreferences,
     private val apiKey: String,
-    private val offeringsCachedObject: InMemoryCachedObject<Offerings> = InMemoryCachedObject(CACHE_REFRESH_PERIOD)
+    private val offeringsCachedObject: InMemoryCachedObject<Offerings> = InMemoryCachedObject(
+        CACHE_REFRESH_PERIOD
+    )
 ) {
-    val legacyAppUserIDCacheKey: String by lazy { "$PREFERENCES_PREFIX$apiKey" }
-    val appUserIDCacheKey: String by lazy { "$PREFERENCES_PREFIX$apiKey.new" }
-    private val attributionCacheKey = "$PREFERENCES_PREFIX.attribution"
-    val tokensCacheKey: String by lazy {"$PREFERENCES_PREFIX$apiKey.tokens"}
+    val legacyAppUserIDCacheKey: String by lazy { "$SHARED_PREFERENCES_PREFIX$apiKey" }
+    val appUserIDCacheKey: String by lazy { "$SHARED_PREFERENCES_PREFIX$apiKey.new" }
+    private val attributionCacheKey = "$SHARED_PREFERENCES_PREFIX.attribution"
+    val tokensCacheKey: String by lazy { "$SHARED_PREFERENCES_PREFIX$apiKey.tokens" }
+    val subscriberAttributesCacheKey: String by lazy { "$SHARED_PREFERENCES_PREFIX$apiKey.subscriberAttributes" }
```

Comment:
why is only one of these private? are the others internal for testing purposes?

Reply:
yeah, I will make all internal